src/vms/pvm/etna-builder/spend-reducers/useUnlockedUTXOs.ts

```diff
@@ -0,0 +1,189 @@
+      .filter(getUsableUTXOsFilter(state));
+
+  // 2. Verify signatures match.
+  const verifiedUsableUTXOs = verifySignaturesMatch(
+    usableUTXOs,
+    (utxo) =>
+      isTransferOut(utxo.output) ? utxo.output : utxo.output.transferOut,
+    state.fromAddresses,
+    state.spendOptions,
+  );
+
+  // 3. Split verified usable UTXOs into AVAX assetId UTXOs and other assetId UTXOs.
+  const { otherVerifiedUsableUTXOs, avaxVerifiedUsableUTXOs } =
+    verifiedUsableUTXOs.reduce<{
+      avaxVerifiedUsableUTXOs: typeof verifiedUsableUTXOs;
+      otherVerifiedUsableUTXOs: typeof verifiedUsableUTXOs;
+    }>(
+      (result, verifiedUsableUTXO) => {
+        if (verifiedUsableUTXO.data.assetId.value() === context.avaxAssetID) {
+          return {
+            ...result,
+            avaxVerifiedUsableUTXOs: [
+              ...result.avaxVerifiedUsableUTXOs,
+              verifiedUsableUTXO,
+            ],
+          };
+        }
+
+        return {
+          ...result,
+          otherVerifiedUsableUTXOs: [
+            ...result.otherVerifiedUsableUTXOs,
+            verifiedUsableUTXO,
+          ],
+        };
+      },
+      { otherVerifiedUsableUTXOs: [], avaxVerifiedUsableUTXOs: [] },
+    );
+
+  // 4. Handle all the non-AVAX asset UTXOs first.
+  for (const { sigData, data: utxo } of otherVerifiedUsableUTXOs) {
+    const utxoInfo = getUtxoInfo(utxo);
+    const remainingAmountToBurn: bigint =
+      state.toBurn.get(utxoInfo.assetId) ?? 0n;
+    const remainingAmountToStake: bigint =
+      state.toStake.get(utxoInfo.assetId) ?? 0n;
+
+    // 4a. If we have already reached the burn/stake amount, there is nothing left to run beyond here.
+    if (remainingAmountToBurn === 0n && remainingAmountToStake === 0n) {
+      continue;
+    }
+
+    // 4b. Add the input.
+    spendHelper.addInput(
+      utxo,
+      new TransferableInput(
+        utxo.utxoId,
+        utxo.assetId,
+        TransferInput.fromNative(utxoInfo.amount, sigData.sigIndicies),
+      ),
+    );
+
+    // 4c. Consume the asset and get the remaining amount.
+    const remainingAmount = spendHelper.consumeAsset(
+      utxoInfo.assetId,
+      utxoInfo.amount,
+    );
+
+    // 4d. If "amountToStake" is greater than 0, add the stake output.
+    // TODO: Implement or determine if needed.
+
+    // 4e. Add the change output if there is any remaining amount.
+    if (remainingAmount > 0n) {
+      spendHelper.addChangeOutput(
+        new TransferableOutput(
+          utxo.assetId,
+          new TransferableOutput(
+            utxo.assetId,
+            new TransferOutput(
+              new BigIntPr(remainingAmount),
+              OutputOwners.fromNative(
+                state.spendOptions.changeAddresses,
+                0n,
+                1,
+              ),
+            ),
+          ),
+        ),
+      );
+    }
+  }
+
+  // 5. Handle AVAX asset UTXOs last to account for fees.
```

Comment:
It seems like we are only using AVAX to cover the fees. How can one use (unlocked) AVAX for staking?